tests/test_hierarchy.cc

```diff
@@ -597,3 +597,65 @@ BOOST_AUTO_TEST_CASE(fast_multiply_transpose_mf)
         BOOST_TEST(fast_matrix->el(i, j) == ref_matrix->el(i, j),
                    tt::tolerance(1e-9));
 }
+
+template <int dim>
+class ConstantMaterialProperty;
+template <int dim>
+class LinearMaterialProperty;
+template <int dim>
+class LinearXMaterialProperty;
+template <int dim>
+class DiscontinuousMaterialProperty;
+
+using material_properties =
+    std::tuple<ConstantMaterialProperty<2>, LinearMaterialProperty<2>,
+               LinearXMaterialProperty<2>, DiscontinuousMaterialProperty<2>>;
+BOOST_AUTO_TEST_CASE_TEMPLATE(test_material_properties, MaterialProperty,
+                              material_properties)
+{
+  dealii::MultithreadInfo::set_thread_limit(1);
+
+  MPI_Comm comm = MPI_COMM_WORLD;
+
+  using DVector = dealii::LinearAlgebra::distributed::Vector<double>;
+  int constexpr dim = mfmg::DealIIMatrixFreeMeshEvaluator<2>::_dim;
```

Comment:
??

Reply:
We do that all over the place in this test.

Reply:
I see two other offenders: https://github.com/ORNL-CEES/mfmg/blob/93e4253727b92d7dbc3d4dd7c743bbaf6c70a9c9/tests/test_hierarchy.cc#L472 and https://github.com/ORNL-CEES/mfmg/blob/93e4253727b92d7dbc3d4dd7c743bbaf6c70a9c9/tests/test_hierarchy.cc#L540

Please change to `int constexpr dim = <DIM>;` and let's merge